src/main/java/jewellery/inventory/service/ProductService.java

```diff
@@ -77,6 +78,10 @@ public ProductReturnResponseDto getProductReturnResponseDto(
     return productMapper.mapToProductReturnResponseDto(sale, product);
   }
 
+  public BigDecimal getProductSalePrice(Product product) {
+    return ProductMapper.calculateTotalPrice(product);
```

Comment:
This method should either return the calculated price or the price from the sale depending on whether the product is sold or not.

Reply:
We use this method only when we create a Sale and for this we shouldn't need to check whether this Product has been sold because we have validations in CreateSale method